### ISSUE TITLE
Latest `1.1.1`release seems to no longer recognize `.jshintrc` in project

### ISSUE THREAD

**Issue by trevordmiller** [AtomLinter/linter-jshint#140]
The latest `1.1.1` release (I updated in Atom today) seems to no longer recognize the `.jshintrc` settings in my projects: I am gettting linter errors/warnings that don't line up with my `.jshintrc` settings (see screenshot below). It has been working for me for months (no code changes) until this last update.

![jshint-issue](https://cloud.githubusercontent.com/assets/5497885/8762915/20e0e228-2d44-11e5-8f39-a181a60f1cd7.png)

**Comment by steelbrain** [comment]
Super-weird, the only update in this release was that it uses atom as the nodejs runtime instead of the system installed one, can you try to hard-code the jshint path in settings and see if it fixes it for you?

**Comment by trevordmiller** [comment]
@steelbrain How can I hard code the `Path of the jshint executable` if it is different for every project? I almost always have a `.jshintrc` at the root of each project I work on...

![jshintrc-path](https://cloud.githubusercontent.com/assets/5497885/8762956/484b5ee6-2d45-11e5-8842-5420b1286eb4.png)

**Comment by steelbrain** [comment]
Dynamic for each project? I don't understand. It's actually the same IMO

**Comment by trevordmiller** [comment]
@steelbrain So, instead of having one master `.jshintrc` in my user directory / atom package that every project points to, I usually have a `.jshintrc` at the root of each project which may or may not be the same...just as you can have other configuration files at the root of a project that may be different for each project on your machine (ie `.editorconfig`, `.gitignore`, `.npmrc`, `package.json` etc.). I believe this is a really common pattern that many GitHub projects use not only for JSHint but for ESLint (`.eslintrc`), JSCS (`.jscsrc`), JSLint (`.jslintrc`) etc. Here are a few examples from popular projects:

- Gulp: https://github.com/gulpjs/gulp
- Grunt: https://github.com/gruntjs/grunt
- jQuery: https://github.com/jquery/jquery

![gulp](https://cloud.githubusercontent.com/assets/5497885/8762989/f4de35ec-2d46-11e5-8e0a-105a0d241bff.png)

![grunt](https://cloud.githubusercontent.com/assets/5497885/8762990/f4df864a-2d46-11e5-83b7-d74dd7d9d43c.png)

![jquery](https://cloud.githubusercontent.com/assets/5497885/8762988/f4d258bc-2d46-11e5-945e-aeb4795c3638.png)

Make sense?

**Comment by steelbrain** [comment]
I know what the problem is. Gonna fix it soon!

**Comment by Glavin001** [comment]
Also experiencing this. `esnext` option as well.

**Comment by peduxe** [comment]
For those having this problem, install version 1.1.0:
```bash
apm uninstall linter-jshint && apm install linter-jshint@1.1.0
```

And then if you have any problem with the executable (when linting), add this to your atom init file:

```javascript
process.env.PATH = ["/usr/local/bin", process.env.PATH].join(":")
```

**Comment by trevordmiller** [comment]
@peduxe Thanks for the tip. Reverting back and adding the the environment path to my init file worked great for me.

**Comment by steelbrain** [comment]
Thanks @peduxe :clap:

**Comment by adambuczynski** [comment]
Just wanted to raise this issue as well, as suddenly my node files were throwing jshint errors. Good to see you're already on it, thanks!

**Comment by nathanbuchar** [comment]
Yep, same here! Looks like it's already being tackled—thanks! :-)

**Comment by giannisp** [comment]
Same issue here, until the update, `apm uninstall linter-jshint && apm install linter-jshint@1.1.0` worked just fine for me.

**Comment by steelbrain** [comment]
I have tracked it down to this line
```coffee
return helpers.execNode(executablePath, parameters, {stdin: text}).then (output) ->
```
if you change it to 
```coffee
return helpers.exec(executablePath, parameters, {stdin: text}).then (output) ->
```
it works, which means that detection is working via global nodejs and not working via electron.

**Comment by steelbrain** [comment]
The error seems to be in `node_modules/jshint/src/cli.js` line `615`, the value of config is different if run via normal iojs or electron.

**Comment by joefitzgerald** [comment]
@steelbrain I'm not sure that modifying the environment in the atom init file is the right solution here. For one thing, it'll break the detection logic in https://atom.io/packages/environment should we decide to integrate it.

**Comment by steelbrain** [comment]
@joefitzgerald but it should still work this way, for users who don't have nodejs installed at all.

**Comment by joefitzgerald** [comment]
I'm saying we shouldn't be instructing people to add `/usr/local/bin` to their path in atom's init file. I'm not commenting on other aspects of your solution.

**Comment by steelbrain** [comment]
@joefitzgerald for `linter-jshint`, users don't have to install **anything**, it's bundled with this package, and uses electron as it's node runtime :wink:

**Comment by joefitzgerald** [comment]
Consider consuming the service provided by environment, and using that for the environment that you pass to execNode. If you don't allow the caller of execNode to specify the environment then you need to make that configurable.

**Comment by joefitzgerald** [comment]
But yes, I understand that jshint is bundled and the embedded node that Atom uses should suffice. My comments simply apply to ensure use of global nose (if required) will work.

**Comment by steelbrain** [comment]
We don't really need node, **anywhere** in this project :sweat_smile:
